src/librustc_lexer/src/lib.rs

```diff
@@ -537,26 +537,30 @@ impl Cursor<'_> {
 
     fn single_quoted_string(&mut self) -> bool {
         debug_assert!(self.prev() == '\'');
-        // Parse `'''` as a single char literal.
-        if self.nth_char(0) == '\'' && self.nth_char(1) == '\'' {
+        // Check if it's a one-symbol literal.
+        if self.second() == '\'' && self.first() != '\\' {
             self.bump();
+            self.bump();
+            return true;
         }
+
+        // Literal has more than one symbol.
+
         // Parse until either quotes are terminated or error is detected.
-        let mut first = true;
         loop {
             match self.first() {
-                // Probably beginning of the comment, which we don't want to include
-                // to the error report.
-                '/' if !first => break,
-                // Newline without following '\'' means unclosed quote, stop parsing.
-                '\n' if self.nth_char(1) != '\'' => break,
-                // End of file, stop parsing.
-                EOF_CHAR if self.is_eof() => break,
```

Comment:
Not sure why the order of `match` arms was changed here.

Reply:
Well, I had two-level motivation here:
1. I ordered match arms depending on the termination level (first match has `return`, then go exceptional cases with `break`, then go char-skipping arms (escaped char and any other char).
2. I thought that it's a bit more readable to have the normal exit condition to be the first match arm.